lua/framework/lua_app.cc

```diff
@@ -53,115 +40,118 @@ LuaApp::InitPetSc(int argc, char** argv)
 int
 LuaApp::Run(int argc, char** argv)
 {
-  InitPetSc(argc, argv);
-  opensn::Initialize();
-  console.PostMPIInfo(opensn::mpi_comm.rank(), opensn::mpi_comm.size());
+  if (opensn::mpi_comm.rank() == 0)
+  {
+    std::cout << opensn::name << " version " << GetVersionStr() << "\n"
+              << Timer::GetLocalDateTimeString() << " Running " << opensn::name << " with "
+              << opensn::mpi_comm.size() << " processes.\n"
+              << opensn::name << " number of arguments supplied: " << argc - 1 << "\n" << std::endl;
+  }
 
-  opensn::log.Log() << opensn::name << " version " << GetVersionStr();
-  opensn::log.Log() << Timer::GetLocalDateTimeString() << " Running " << opensn::name << " with "
-                    << opensn::mpi_comm.size() << " processes.";
-  opensn::log.Log() << opensn::name << " number of arguments supplied: " << argc - 1;
-  opensn::log.LogAll();
-  console.FlushConsole();
+  int error_code = ParseArguments(argc, argv);
 
-  int error_code = 0;
-  ParseArguments(argc, argv);
-  if (not termination_posted_)
+  if (!error_code)
   {
+    InitPetSc(argc, argv);
+    opensn::Initialize();
+    console.PostMPIInfo(opensn::mpi_comm.rank(), opensn::mpi_comm.size());
+    console.FlushConsole();
+
     if (sim_option_interactive_)
       error_code = RunInteractive(argc, argv);
     else
       error_code = RunBatch(argc, argv);
-  }
 
-  opensn::Finalize();
-  PetscFinalize();
+    opensn::Finalize();
+    PetscFinalize();
 
-  opensn::log.Log() << "Elapsed execution time: " << program_timer.GetTimeString();
-  opensn::log.Log() << Timer::GetLocalDateTimeString() << " " << opensn::name
-                    << " finished execution.";
+    if (opensn::mpi_comm.rank() == 0)
+    {
+      std::cout << "\n"
+                << Elapsed execution time: " << program_timer.GetTimeString() << "\n"
+                << Timer::GetLocalDateTimeString() << " " << opensn::name
+                << " finished execution." << std::endl;
+    }
+  }
 
   return error_code;
 }
 
-void
+int
 LuaApp::ParseArguments(int argc, char** argv)
 {
-  bool input_file_found = false;
-  for (int i = 1; i < argc; i++)
+  try
   {
-    std::string argument(argv[i]);
+    cxxopts::Options options("opensn", "");
```

Comment:
Ideally, this should be `LowerCase(opensn::name)`, where `LowerCase` is a function that converts a string into a lowercase string. I don't think there is one in OpenSn. And shame on C++ that it still does not have it (!). 

I usually have one like this:
```
std::string
LowerCase(const std::string & name)
{
    std::string lower(name);
    std::transform(lower.begin(), lower.end(), lower.begin(), ::tolower);
    return lower;
}
```

This could be used later on for case-insensitive string comparisons. I am saying this to prevent hard-coding the transformation in place...

Reply:
Fixed.